Include/internal/pycore_dtoa.h

```diff
@@ -17,7 +19,22 @@ PyAPI_FUNC(void) _Py_dg_freedtoa(char *s);
 PyAPI_FUNC(double) _Py_dg_stdnan(int sign);
 PyAPI_FUNC(double) _Py_dg_infinity(int sign);
 
+#define Kmax 7
+
+typedef uint32_t ULong;
+typedef int32_t Long;
+typedef uint64_t ULLong;
```

Comment:
Many C files include directly or indirectly pycore_interp.h which includes pycore_dtoa.h. I would prefer to not pollule their namespaces with names like Kmax, ULong, Long, UULong which can clash with other names.

I see different options:

* Don't fix dtoa.c for now
* Add a prefix to all these names in the header file (ex: "_PyDtoa_") and create aliases in dtoa.c (ex: `#define Kmax _PyDtoa_Kmax`).
* Add an API like PyInterpreterState.dict to have per-interpreter variables: the freelist would be dynamically allocated on the heap memory, and stored as a raw pointer in PyInterpreterState. Or if you don't want to add a whole API, just expose the cache as a "void *" in the structure, and add init/fini functions to allocate/free this structure (free list).

Reply:
thanks for your review. I chose the second option to fix this problem.